src/storage/txn/actions/get_old_value.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+use crate::storage::mvcc::reader::seek_for_valid_write;
+use crate::storage::mvcc::{MvccTxn, Result as MvccResult};
+use crate::storage::Snapshot;
+use txn_types::{Key, OldValue, Write};
+
+/// Read the old value for key for CDC.
+/// `prev_write` stands for the previous write record of the key
+/// it must be read in the caller and be passed in for optimization
+pub fn get_old_value<S: Snapshot>(
+    txn: &mut MvccTxn<S>,
+    key: &Key,
+    prev_write: Option<Write>,
+) -> MvccResult<OldValue> {
+    match prev_write {
+        Some(w) => {
+            let valid_write = if w.write_type.has_old_value() {
+                Some(w)
+            } else {
+                last_valid_write(txn, key)?
+            };
+            Ok(valid_write
+                .map(|w| OldValue::Value {
+                    short_value: w.short_value,
+                    start_ts: w.start_ts,
+                })
+                .unwrap_or(OldValue::None))
```

Comment:
Can you impl `From<Option<Write>>` for Write, so that this can be `valid_write.into()`?